### ISSUE TITLE
Desktop: Form elements not behaving according to standard HTML form elements

### ISSUE THREAD

**Issue by hhonshuku** [kobotoolbox/enketo-express#126]
A user might expect the familiar HTML form elements behavior.
- User should be able to navigate all the HTML elements by [TAB] key, and should be able to select/execute the element by pressing [Space].
- An item of an option list can be selected by typing the first few letters, i.e., when the list is for States, typing 'mas' should select Massachusetts
- A modal window such as Successful Submission Message Window should be closed by [Enter/Return] key
- Enketo does not allow [TAB] to navigate to Checkbox and Radio Button blocks
- Enketo uses [Enter/Return] instead of [Space] to select a highlighted item in the option list, which I think is against the standard expectation

**Comment by MartijnR** [comment]
https://bugzilla.mozilla.org/show_bug.cgi?id=605985 (the remaining issue for radio buttons/ checkboxes alternative). Not sure about printability, but we could keep current solution in print view otherwise :(

**Comment by MartijnR** [comment]
first spend a couple of hours to check feasibility of alternative approach across browsers (particularly firefox).

**Comment by MartijnR** [comment]
http://codepen.io/MartijnR/pen/XpexOR

**Comment by MartijnR** [comment]
conclusion: I think we can go ahead using a background image (same as background used for current icons), and accept that FF will revert to regular checkboxes/radiobutton, until they fix their bug.

**Comment by MartijnR** [comment]
Will also go-ahead with file inputs (both are ESRI-funded improvements for Enketo Core)

**Comment by MartijnR** [comment]
Remaining: 

* [ ] file inputs 
* [ ] autocomplete for pulldown (this will be a widget, but worth considering to make this the default)
* [ ] pulldown option selection with spacebar (investigate)
* [ ] closing a modal dialog (investigate)
